server/src/main/java/io/druid/segment/realtime/plumber/RealtimePlumber.java

```diff
@@ -450,19 +482,27 @@ public void doRun()
               }
 
               QueryableIndex index = IndexIO.loadIndex(mergedFile);
-
-              DataSegment segment = dataSegmentPusher.push(
-                  mergedFile,
-                  sink.getSegment().withDimensions(Lists.newArrayList(index.getAvailableDimensions()))
-              );
-
-              segmentPublisher.publishSegment(segment);
-
-              if (!isPushedMarker.createNewFile()) {
-                log.makeAlert("Failed to create marker file for [%s]", schema.getDataSource())
-                   .addData("interval", sink.getInterval())
-                   .addData("partitionNum", segment.getShardSpec().getPartitionNum())
-                   .addData("marker", isPushedMarker)
+              log.info("Pushing [%s] to deep storage", sink.getSegment().getIdentifier());
+              try {
+                DataSegment segment = dataSegmentPusher.push(
+                    mergedFile,
+                    sink.getSegment().withDimensions(Lists.newArrayList(index.getAvailableDimensions()))
+                );
+                log.info("Inserting [%s] to the metadata store", sink.getSegment().getIdentifier());
+                segmentPublisher.publishSegment(segment);
+
+                if (!isPushedMarker.createNewFile()) {
+                  log.makeAlert("Failed to create marker file for [%s]", schema.getDataSource())
+                     .addData("interval", sink.getInterval())
+                     .addData("partitionNum", segment.getShardSpec().getPartitionNum())
+                     .addData("marker", isPushedMarker)
+                     .emit();
+                }
+              }
+              catch (Throwable e) {
```

Comment:
Why Throwable instead of Exception? Usually we don't catch and toss away Throwables.

Also, why did you add this try block here, as opposed to letting the catch block below handle exceptions during pushing or publishing?